job-runner/chain/traverser.go

```diff
@@ -199,9 +199,21 @@ JOB_REAPER:
 			break
 		}
 
-		// If the job did not complete successfully, then ignore subsequent jobs.
-		// For example, if job B of A -> B -> C  fails, then C is not ran.
-		if doneJ.State != proto.STATE_COMPLETE {
+		// t.runJobs() sets every job that isn't COMPLETE to be FAIL.
+		if doneJob.State != proto.STATE_COMPLETE {
+			// If the job is part of a retryable sequence, retry that sequence.
+			if t.chain.CanRetrySequence(doneJob) {
+				err := t.retrySequence(doneJob)
```

Comment:
Since `retrySequence` cannot fail (return error), let's make it not return anything. The error handling here is superfluous. Without it, the block is much cleaner imho:
```go
jLogger.Info("job did not complete successfully. retrying sequence.")
t.retrySequence(doneJob) // re-enqueue first job of failed sequence
continue JOB_REAPER
```

Reply:
good point. i must have been returning an error before and removed it. thank you